apps/base/utils.py

```diff
@@ -18,20 +18,18 @@ def paginated_queryset(queryset, request):
     return (paginator, result_page)
 
 
-@deconstructible
 class RandomFileName(object):
     def __init__(self, path):
         self.path = path
 
     def __call__(self, instance, filename):
         extension = os.path.splitext(filename)[1]
         if 'id' in self.path and instance.pk:
-            self.path = self.path.format(id=instance.pk)
-        elif 'id' not in self.path and instance.pk:
-            path = "submission_files/submission_{id}"
-            self.path = path.format(id=instance.pk)
+            path = self.path.format(id=instance.pk)
+        else:
```

Comment:
What if we define path as a variable above the if condition and then when the if condition is true, we can simply change it. This way it will be more better. Let me know what are your thoughts over this.